src/lib/com/yetanalytics/xapipe/filter.clj

```diff
@@ -154,28 +154,23 @@
 (defn get-state-key
   "Given a statement, return a state key if possible, or nil"
   [statement]
-  (let [?reg (get-in statement ["context" "registration"])
-        ?subreg (get-in statement ["context" "extensions" per/subreg-iri])]
+  (let [?reg (get-in
+              statement
+              ["context"
+               "registration"])
+        ?subreg (get-in
+                 statement
+                 ["context"
+                  "extensions"
+                  "https://w3id.org/xapi/profiles/extensions/subregistration"])]
     (cond
       (and ?reg ?subreg) [?reg ?subreg]
       ?reg               ?reg
       :else              nil)))
```

Comment:
Some things
- `?subreg` will be a vector of maps, not a UUID like you'd want.
- There is some validation you ought to do (e.g. ensuring that you don't have subregistration when there's no registration)
See https://github.com/yetanalytics/project-persephone/blob/main/src/main/com/yetanalytics/persephone/utils/statement.cljc#L48-L96 for how persephone extracts the registration and subregistration from a Statement.

Reply:
> ?subreg will be a vector of maps, not a UUID like you'd want.

Really? the extension value is a map? What's the path inside then?

Reply:
Also see this: https://github.com/yetanalytics/project-persephone/blob/main/src/main/com/yetanalytics/persephone.cljc#L359-374

Reply:
It's insane, but each subregistration object has `profile` and `subregistration` key, and you'd have to filter over `profile` with your Profile ID to get the `subregistration` you want.

Reply:
> There is some validation you ought to do ...

Nope, not for this use case. As you can see in the `cond` it only passes two if both are found (though I could add not-empty checks there). If there's just a subreg it isn't the concern of `xapipe`. Not validating, filtering via matching

Reply:
> It's insane, but each subregistration object has profile and subregistration key, and you'd have to filter over profile with your Profile ID to get the subregistration you want

That is legitimately crazy, thanks for letting me know.

Reply:
It would be cool if either pan or persephone exposed a function for this in their public API, though I'm happy to just throw one in here. For filtering it's a major optimization for me to just not try to match if it doesn't find a reg/reg+subreg tuple

Reply:
You could use the `util.statement` namespace, which exposes some of these functions.

Reply:
Yeah if those aren't too subject to change, that sounds great. More red!

Reply:
I also created a small PR to expose the `get-subregistration-id` function. From there you can easily create your own reg-subreg pairs or similar: https://github.com/yetanalytics/project-persephone/pull/28

Reply:
See https://github.com/yetanalytics/xapipe/pull/61/commits/26c032c7cd7767bdc0e03ceee97eff55112f508e